routing/payment_lifecycle.go

```diff
@@ -597,9 +597,6 @@ func (p *shardHandler) collectResult(attempt *channeldb.HTLCAttemptInfo) (
 
 	case <-p.router.quit:
 		return nil, ErrRouterShuttingDown
-
```

Comment:
Can we get a more complete PR body here describing the issue and how removing this select statement resolve it? 

(glanced at an earlier version of this PR, and curious to track how our understanding of the fix has evolved)